core/src/main/java/org/wso2/msf4j/internal/MSF4JMessageProcessor.java

```diff
@@ -70,38 +92,105 @@ public MSF4JMessageProcessor(String channelId, MicroservicesRegistryImpl microse
-            } catch (HandlerException e) {
-                handleHandlerException(e, carbonCallback);
-            } catch (InvocationTargetException e) {
-                Throwable targetException = e.getTargetException();
-                if (targetException instanceof HandlerException) {
-                    handleHandlerException((HandlerException) targetException, carbonCallback);
-                } else {
-                    handleThrowable(currentMicroservicesRegistry, targetException, carbonCallback, request);
+            //Identify the protocol name before doing the processing
+            String protocolName = (String) carbonMessage.getProperty(Constants.PROTOCOL);
+
+            if (Constants.HTTP_PROTOCOL_NAME.equalsIgnoreCase(protocolName)) {
+                MicroservicesRegistryImpl currentMicroservicesRegistry =
+                        DataHolder.getInstance().getMicroservicesRegistries()
+                                .get(carbonMessage.getProperty(MSF4JConstants.CHANNEL_ID));
+                Request request = new Request(carbonMessage);
+                request.setSessionManager(currentMicroservicesRegistry.getSessionManager());
+                Response response = new Response(carbonCallback, request);
+                try {
+                    dispatchMethod(currentMicroservicesRegistry, request, response);
+                } catch (HandlerException e) {
+                    handleHandlerException(e, carbonCallback);
+                } catch (InvocationTargetException e) {
+                    Throwable targetException = e.getTargetException();
+                    if (targetException instanceof HandlerException) {
+                        handleHandlerException((HandlerException) targetException, carbonCallback);
+                    } else {
+                        handleThrowable(currentMicroservicesRegistry, targetException, carbonCallback, request);
+                    }
+                } catch (InterceptorException e) {
+                    log.warn("Interceptors threw an exception", e);
+                    // TODO: improve the response
+                    carbonCallback.done(HttpUtil.createTextResponse(
+                            javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
+                            HttpUtil.EMPTY_BODY));
+                } catch (Throwable t) {
+                    handleThrowable(currentMicroservicesRegistry, t, carbonCallback, request);
+                } finally {
+                    // Calling the release method to make sure that there won't be any memory leaks from netty
+                    carbonMessage.release();
+                }
+
+            } else if (Constants.WEBSOCKET_PROTOCOL_NAME.equalsIgnoreCase(protocolName)) {
+                EndpointsRegistryImpl endpointsRegistry = EndpointsRegistryImpl.getInstance();
+                PatternPathRouter.RoutableDestination<Object>
+                        routableEndpoint = null;
```

Comment:
Variable declaration can be inside the try block. No need to initialize this as null